File: cpp/Problem70.cpp
/*
 * Problem: 70. Climbing Stairs
 * URL: https://leetcode.com/problems/climbing-stairs/
 * Difficulty: Easy
 * Tags: Array, Dynamic Programming(DP)
 */

class Solution {
public:
	int climbStairs(int n) {
		int results[46];
		results[0] = 1;
		results[1] = 1;
		for (int i = 2; i <= n; i++) {
			results[i] = results[i - 1] + results[i - 2];
		}
		return results[n];
	}
};

File: python/Problem70.py
# Problem: 70. Climbing Stairs
# URL: https://leetcode.com/problems/climbing-stairs/
# Difficulty: Easy
# Tags: Array, Dynamic Programming(DP)

from typing import List

class Solution:
  def climbStairs(self, n: int) -> int:
    results: List[int] = [0] * 46
    results[0] = 1
    results[1] = 1
    
    for i in range(2, 46):
      results[i] = results[i - 1] + results[i - 2]
    
    return results[n]